File: frontend/src/lib/permissions.ts
// CANIT Pulse — Frontend RBAC Permissions Engine
// Maps roles to specific UI capabilities

export type UserRole = "super_admin" | "csm" | "hr" | "employee" | "client" | "admin";

export const ROLE_DISPLAY_NAMES: Record<string, string> = {
  super_admin: "Super Admin",
  csm: "CSM",
  hr: "HR",
  employee: "Employee",
  client: "Client",
  admin: "Super Admin", // Legacy alias
};

// ── Permission Helpers ──────────────────────────────────

export const canCreateClient = (role: string | undefined): boolean =>
  !!role && ["super_admin", "csm", "admin"].includes(role);

export const canEditClient = (role: string | undefined): boolean =>
  !!role && ["super_admin", "csm", "admin"].includes(role);

export const canDeleteClient = (role: string | undefined): boolean =>
  !!role && ["super_admin", "admin"].includes(role);

export const canArchiveClient = (role: string | undefined): boolean =>
  !!role && ["super_admin", "admin"].includes(role);

export const canGenerateReports = (role: string | undefined): boolean =>
  !!role && ["super_admin", "csm", "admin"].includes(role);

export const canAccessSettings = (role: string | undefined): boolean =>
  !!role && ["super_admin", "admin"].includes(role);

export const canManageUsers = (role: string | undefined): boolean =>
  !!role && ["super_admin", "admin"].includes(role);

export const canConnectSocial = (role: string | undefined): boolean =>
  !!role && ["super_admin", "csm", "admin"].includes(role);

export const canAccessClientPortal = (role: string | undefined): boolean =>
  !!role && ["super_admin", "csm", "client", "admin"].includes(role);

export const canViewAllClients = (role: string | undefined): boolean =>
  !!role && ["super_admin", "csm", "hr", "admin"].includes(role);

export const canViewReports = (role: string | undefined): boolean =>
  !!role && ["super_admin", "csm", "hr", "employee", "admin"].includes(role);

export const canViewArchive = (role: string | undefined): boolean =>
  !!role && ["super_admin", "csm", "hr", "employee", "admin"].includes(role);

export const canDeleteReport = (role: string | undefined): boolean =>
  !!role && ["super_admin", "admin"].includes(role);

export const isInternal = (role: string | undefined): boolean =>
  !!role && ["super_admin", "csm", "hr", "employee", "admin"].includes(role);

export const canUseAi = (role: string | undefined): boolean =>
  !!role && ["super_admin", "csm", "hr", "employee", "client", "admin"].includes(role);

File: backend/services/permissions.py
"""
CANIT Pulse — RBAC Permissions Engine

Roles:
  • super_admin — Full platform access (Raj, Sharmu, Jaishree)
  • csm         — Client management, reports, social (no delete/settings/users)
  • hr          — View-only access (dashboard, reports, client info)
  • employee    — View-only access (dashboard, reports, calendar, analytics)
  • client      — Own dashboard only (assigned via Client Access Portal)
  • admin       — Legacy alias → maps to super_admin capabilities
"""


def can_create_client(role: str) -> bool:
    return role in ("super_admin", "csm", "admin")

def can_edit_client(role: str) -> bool:
    return role in ("super_admin", "csm", "admin")

def can_delete_client(role: str) -> bool:
    return role in ("super_admin", "admin")

def can_archive_client(role: str) -> bool:
    return role in ("super_admin", "admin")

def can_generate_reports(role: str) -> bool:
    return role in ("super_admin", "csm", "admin")

def can_access_settings(role: str) -> bool:
    return role in ("super_admin", "admin")

def can_manage_users(role: str) -> bool:
    return role in ("super_admin", "admin")

def can_connect_social(role: str) -> bool:
    return role in ("super_admin", "csm", "admin")

def can_access_client_portal(role: str) -> bool:
    return role in ("super_admin", "csm", "client", "admin")

def can_view_all_clients(role: str) -> bool:
    return role in ("super_admin", "csm", "hr", "admin")

def can_view_reports(role: str) -> bool:
    return role in ("super_admin", "csm", "hr", "employee", "admin")

def can_delete_report(role: str) -> bool:
    return role in ("super_admin", "admin")

def is_internal(role: str) -> bool:
    """Returns True for all internal team roles (NOT client)."""
    return role in ("super_admin", "csm", "hr", "employee", "admin")

def can_use_ai(role: str) -> bool:
    return role in ("super_admin", "csm", "hr", "employee", "client", "admin")
